lib/gollum-lib/page.rb

```diff
@@ -17,29 +17,25 @@ class Page
     # Returns a Page
     attr_accessor :parent_page
 
-
-    # Checks a filename against the registered markup extensions
+    # Find the Symbol format for the file extension
     #
-    # filename - String filename, like "Home.md"
+    # ext - String file extension, like ".md"
     #
-    # Returns e.g. ["Home", :markdown], or [] if the extension is unregistered
-    def self.parse_filename(filename)
-      return [] unless filename =~ /^(.+)\.([a-zA-Z]\w*)$/i
-      pref, ext = Regexp.last_match[1], Regexp.last_match[2]
-
+    # Returns the Symbol format associated with ext
+    def self.parse_extension(ext)
       Gollum::Markup.formats.each_pair do |name, format|
```

Comment:
This is still not very efficient: we have to loop through all the registered markups and check whether the defined regexp matches `ext`. I don't actually understand why we don't simply register valid extensions as strings rather than regexps, so that we could just do a `VALID_EXTENSIONS.include?(ext)` here. Perhaps this is good further refactoring for `5.x`?